### PR TITLE
[8.1] Malformed CIDs do not pass URL validation.

### PR DESCRIPTION
**Pull request by vmcj**
Documentation needs 400 for all inputs, so we loose currently a more
specific message. Alternative is duplication of code.

### CODE REVIEW

**Comment by nickygerritsen** [comment]
Could we not limit what is allowed as ID's? See https://ccs-specs.icpc.io/draft/contest_api#json-attribute-types what is allowed:
> `[a-zA-Z0-9_.-]` of length at most 36 and not starting with a `-` (dash) or `.` (dot) or ending with a `.` (dot).

**Comment by mvr320** [comment]
> 
> 
> Could we not limit what is allowed as ID's? See https://ccs-specs.icpc.io/draft/contest_api#json-attribute-types what is allowed:
> 
> > `[a-zA-Z0-9_.-]` of length at most 36 and not starting with a `-` (dash) or `.` (dot) or ending with a `.` (dot).

I think that is a good idea but:
- Still if someone provides "wrong" data nginx will return a 400, the problem is that the fuzzer provides this wrong data and receives a 400 which was never stated.
- The documentation should cover all the states AND declare what we expect (the regex you gave).

Overall I think this is also just a mistake with OpenAPI, I think it should always be implicit that when you provide something wrong you're going to get a 400.

**Comment by nickygerritsen** [comment]
Alternative: can we write something (some event subscriber or similar) that adds 400 to all calls?  I remember I did something similar in the past

**Comment by mvr320** [comment]
> 
> 
> Alternative: can we write something (some event subscriber or similar) that adds 400 to all calls? I remember I did something similar in the past

I think we can define it in 1 file, you will still lose the specific message, so is your problem with the duplication of code or with the problem that we lose a more specific error message?

**Comment by nickygerritsen** [comment]
> > Alternative: can we write something (some event subscriber or similar) that adds 400 to all calls? I remember I did something similar in the past
> 
> I think we can define it in 1 file, you will still lose the specific message, so is your problem with the duplication of code or with the problem that we lose a more specific error message?

I like less copy paste. What specific messages do we have?

File: webapp/src/Controller/API/ContestController.php [hunk 1]
```diff
@@ -237,10 +238,6 @@ public function bannerAction(Request $request, string $id): Response
      *     description="Contest start time changed successfully",
      * )
      * @OA\Response(
-     *     response="400",
-     *     description="Invalid input data"
```
Comment (mvr320) [inline webapp/src/Controller/API/ContestController.php:241]: @nickygerritsen this one.

Comment (nickygerritsen) [reply webapp/src/Controller/API/ContestController.php:241]: Yeah I don't care about that one at all 😄

**Comment by thijskh** [comment]
I don’t see any change in Nginx config?

**Comment by mvr320** [comment]
> I don’t see any change in Nginx config?

nginx+apache are not wrong,

URL:
domjudge/api/contests/% should throw 400, the problem is just that we don't declare that that can happen.

**Comment by thijskh** [comment]
So what does the nginx in the title of this PR refer to if nginx is not involved?

**Comment by mvr320** [comment]
> So what does the nginx in the title of this PR refer to if nginx is not involved?

That I initially had this on nginx, but that it should be fixed in our API spec.

**Comment by vmcj** [comment]
Merged for the DJ80 release and @nickygerritsen his suggestion will be implemented for DJ81

**Pull request merged**